packages/contractkit/src/wrappers/MetaTransactionWallet.ts

```diff
@@ -0,0 +1,279 @@
+
+export interface MTWTransaction {
+  destination: Address
+  value?: BigNumber.Value
+  data?: string
+}
+
+export interface MTWMetaTransaction extends MTWTransaction {
+  nonce: number
+}
+
+export interface MTWSignedMetaTransaction extends MTWTransaction {
+  nonce: number
+  signature: Signature
+}
+
+/**
+ * Class that wraps the MetaTransactionWallet
+ */
+export class MetaTransactionWalletWrapper extends BaseWrapper<MetaTransactionWallet> {
+  /**
+   * Execute a transaction originating from the MTW
+   * Reverts if the caller is not a signer
+   * @param tx a MTWTransaction
+   */
+  executeTransaction(tx: MTWTransaction) {
+    return toTransactionObject(
+      this.kit,
+      this.contract.methods.executeTransaction(
+        tx.destination,
+        numericToHex(tx.value),
+        tx.data || '0x'
+      )
+    )
+  }
+
+  /**
+   * Execute a batch of transactions originating from the MTW
+   * Reverts if the caller is not a signer
+   * @param txs An array of MTWTransactions
+   */
+  executeTransactions(txs: MTWTransaction[]) {
+    return toTransactionObject(this.kit, this.buildExecuteTransactionsTx(txs))
+  }
+
+  /**
+   * Builds the TransactionObject for executeTransactions
+   * Reverts if the caller is not a signer
+   * @param txs An array of MTWTransactions
+   */
+  buildExecuteTransactionsTx(txs: MTWTransaction[]) {
+    const destinations = txs.map((tx) => tx.destination)
+    const values = txs.map((tx) => numericToHex(tx.value))
+    const callData = ensureLeading0x(txs.map((tx) => trimLeading0x(tx.data || '0x')).join(''))
+    const callDataLengths = txs.map((tx) => trimLeading0x(tx.data || '0x').length / 2)
+
+    return this.contract.methods.executeTransactions(
+      destinations,
+      values,
+      callData,
+      callDataLengths
+    )
+  }
+
+  /**
+   * Builds a MetaTransaction that wraps a series of Transactions
+   * Reverts if the caller is not a signer
+   * @param txs An array of MTWTransactions
+   * @returns MTWMetaTransaction
+   */
+  buildExecuteTransactionsWrapperTx(txs: MTWTransaction[]): MTWTransaction {
+    const executeTransactionsTx = this.buildExecuteTransactionsTx(txs)
+
+    return {
+      destination: this.address,
+      value: new BigNumber(0),
+      data: executeTransactionsTx.encodeABI(),
+    }
+  }
+
+  /**
+   * Execute a signed meta transaction
+   * Reverts if meta-tx signer is not a signer for the wallet
+   * @param mtx a MTWSignedMetaTransaction
+   */
+  executeMetaTransaction(mtx: MTWSignedMetaTransaction) {
+    return toTransactionObject(
+      this.kit,
+      this.contract.methods.executeMetaTransaction(
+        mtx.destination,
+        numericToHex(mtx.value),
+        mtx.data || '0x',
+        mtx.signature.v,
+        mtx.signature.r,
+        mtx.signature.s
+      )
+    )
+  }
+
+  /**
+   * Signs a meta transaction as EIP712 typed data
+   * @param mtx a MTWTransaction
+   * @returns smtx a MTWSignedMetaTransaction
+   */
+  async signMetaTransaction(
```

Comment:
Consider adding a `signMetaTransactions` function, that batches the transactions into a single `executeTransactions` call and then signs that as a meta transaction.